src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -17,6 +17,400 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+            CompileAndVerify(compilation, expectedOutput:
+@"target default value
+arg default value
+caller target value
+callerTargetExp
+caller target value
+callerTargetExp
+target default value
+arg default value
+caller target value
+callerTargetExp");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestCallerArgumentExpressionWithDifferentParametersReferringToEachOther()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    public static void Main()
+    {
+        M();
+        M(""param1_value"");
+        M(param1: ""param1_value"");
+        M(param2: ""param2_value"");
+        M(param1: ""param1_value"", param2: ""param2_value"");
+        M(param2: ""param2_value"", param1: ""param1_value"");
+    }
+
+    static void M([CallerArgumentExpression(""param2"")] string param1 = ""param1_default"", [CallerArgumentExpression(""param1"")] string param2 = ""param2_default"")
+    {
+        Console.WriteLine($""param1: {param1}, param2: {param2}"");
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput:
+@"param1: param1_default, param2: param2_default
+param1: param1_value, param2: ""param1_value""
+param1: param1_value, param2: ""param1_value""
+param1: ""param2_value"", param2: param2_value
+param1: param1_value, param2: param2_value
+param1: param1_value, param2: param2_value");
+        }
+
+        // PROTOTYPE(caller-expr): Should caller argument expression be given an argument that's compiler generated?
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpressionIsCallerMember()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+C.M();
+
+public static class C
+{
+    public static void M(
+        [CallerMemberName] string callerName = """",
+        [CallerArgumentExpression(""callerName"")] string argumentExp = """")
+    {
+        Console.WriteLine(callerName);
+        Console.WriteLine(argumentExp);
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe);
+            CompileAndVerify(compilation, expectedOutput: "<Main>$");
+        }
+
+        // PROTOTYPE(caller-expr): Should caller argument expression be given an argument that's compiler generated?
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpressionIsReferingToItself()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+C.M();
+C.M(""value"");
+
+public static class C
+{
+    public static void M(
+        [CallerArgumentExpression(""p"")] string p = """")
```

Comment:
>"""" [](start = 53, length = 4)

Consider making this not an empty string, so we can see it better (or lack of it better) in the output.

Reply:
Thanks. Will fix that in all tests. Also found that I'm missing VerifyDiagnostics in many tests, will fix that too.

Reply:
Fixed.